### DISCUSSION TITLE
Is there a way of saying "For all other Triggers"?

### DISCUSSION THREAD

**Discussion by jamescurran** [dotnet-state-machine/stateless#523] [Q&A]
I'm trying to use the package to parse a string.  Certain characters start specific commands; these are obviously triggers.  Every other character just needs to be added to the string being built.  Is there a way, when configuring states, to say "for all other triggers, do this ..." ???

Example of what I'm trying:
input:
  "This is plain text. !{B;TC:Red}This is Bold, Red text!#, text, !{C:Blue}Text with Blue background!!# boring text"
output:
   This is plain text. This is Bold, Red text, text, Text with Blue background! boring text
   [Bold: 20-41]
   [Red:20-41]
   [Blue Background:5-75]

**Comment by jamescurran** [comment]
Here are the states and triggers I've planned.  In the below, underscore ("_") means "All other characters":
Text
	! => Bang
	_ => save character => Text

Bang
	{ => (Create command object, save start position) => Command
	# => (save end position, queue command object) => Text
	! => (Save bang) => Bang
	_ => (save bang, save character) => Text

Command
	} => (push command object) => Text
	; => (push command object) => Command
	: => Parameter
	_ => (add to command) => Command

Parameter
	} => (push command object) => Text
	; => (push command object) => Command
	_ => (add to parameter) => Parameter

**Comment by mclift** [comment]
Assuming your state machine is based on `char` as the trigger type, would an `OnUnhandledTrigger` configuration work for this?

**Reply by jamescurran** [reply]
Possibly, but it'll be a bit kludgy.  I need it to go to a new state on the unhandled triggger.

The way I handled it was to add a lexing step:

```
		private Trigger Lex(char ch)
		{
			_currentChar = ch;
			if (Char.IsDigit(ch))
				return Trigger.Digit;

			return ch switch
			{
				'!' => Trigger.Bang,
				'{' => Trigger.OpenBrace,
				'}' => Trigger.CloseBrace,
				':' => Trigger.Colon,
				';' => Trigger.Semi,
				'#' => Trigger.Hash,
				'|' => Trigger.Pipe,
				'_' => Trigger.Underscore,
				'[' => Trigger.OpenBracket,
				']' => Trigger.CloseBracket,
				'~' => Trigger.Tilde,
				_ => Trigger.Char
			};
		}
```

And then add a bunch of similar `Permit`s on many State definitions:
```
			_stringWithCodes.Configure(State.Initial)
				.OnEntry(Reset)
				.Permit(Trigger.Bang, State.Bang)
				.Permit(Trigger.OpenBrace, State.Replacement)
				.Permit(Trigger.Underscore, State.NextCell)
				.Permit(Trigger.EndOfString, State.CompleteParse)
				.Permit(Trigger.Pipe, State.AddLineBreak)
				.Permit(Trigger.Tilde, State.Literal)
				.Permit(Trigger.Char, State.Text)
				.Permit(Trigger.Digit, State.Text)
				.Permit(Trigger.CloseBrace, State.Text)
				.Permit(Trigger.Semi, State.Text)
				.Permit(Trigger.Colon, State.Text)
				.Permit(Trigger.Hash, State.Text)
				.PermitReentry(Trigger.Reset)
                                 ;
```

Seven of my state configurations have 6 lines in the form `.Permit(Trigger.xxxxx, State.Text)`
